runtime-sdk/src/crypto/multisig/mod.rs

```diff
@@ -0,0 +1,82 @@
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+use crate::crypto::signature::{PublicKey, Signature};
+
+#[cfg(test)]
+mod test;
+
+/// Error.
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("invalid config")]
+    InvalidConfig,
+    #[error("insufficient weight")]
+    InsufficientWeight,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Signer {
+    pub public_key: PublicKey,
+    pub weight: u64,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Config {
+    pub signers: Vec<Signer>,
+    pub threshold: u64,
+}
+
+impl Config {
+    pub fn verify(&self) -> Result<(), Error> {
+        if self.threshold == 0 {
+            return Err(Error::InvalidConfig);
+        }
+        let mut total: u64 = 0;
+        for (i, signer) in self.signers.iter().enumerate() {
+            if self
+                .signers
+                .iter()
+                .take(i)
+                .any(|other_signer| signer.public_key == other_signer.public_key)
```

Comment:
I don't know how big these get. we might later prefer to use a hash. would have to figure out how that works on the secp256k1 side.

Reply:
There should be parameters to limit these, related to #120 (this one is for transaction signers, but the issue is similar). Also gas costs should probably depend on the number of signers as it is more costly to verify multiple signatures.

Reply:
added these limits. haven't added gas cost though